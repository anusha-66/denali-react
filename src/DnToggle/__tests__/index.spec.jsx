/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnToggle } from '../../index'; // src/index.tsx

describe('DnToggle Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnToggle>
                <DnToggle.Item isActive>Toggle1</DnToggle.Item>
                <DnToggle.Item>Toggle2</DnToggle.Item>
            </DnToggle>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render with no class names', () => {
        expect.assertions(1);

        const rendered = render(
            <DnToggle>
                <DnToggle.Item>Toggle1</DnToggle.Item>
                <DnToggle.Item>Toggle2</DnToggle.Item>
            </DnToggle>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnToggle isSmall isInverse className="additional--class">
                <DnToggle.Item isActive>Toggle1</DnToggle.Item>
                <DnToggle.Item isActive={false} isDisabled>
                    Toggle2
                </DnToggle.Item>
                <DnToggle.Item isActive={false}>Toggle3</DnToggle.Item>
            </DnToggle>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with radio item', () => {
        expect.assertions(1);

        const rendered = render(
            <DnToggle isSmall isInverse isRadioToggle>
                <DnToggle.RadioItem
                    id="Radio1"
                    name="sample"
                    radioValue="Radio1"
                    defaultChecked
                    isDisabled={false}
                    radioLabel="Radio 1"
                />
                <DnToggle.RadioItem
                    id="Radio2"
                    name="sample"
                    radioValue="Radio2"
                    defaultChecked={false}
                    isDisabled={false}
                    radioLabel="Radio 2"
                />
                <DnToggle.RadioItem
                    id="Radio3"
                    name="sample"
                    radioValue="Radio3"
                    defaultChecked={false}
                    isDisabled={false}
                    radioLabel="Radio 3"
                />
            </DnToggle>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
