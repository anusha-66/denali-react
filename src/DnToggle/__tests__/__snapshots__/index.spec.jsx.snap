// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnToggle Tests should render correctly with radio item 1`] = `
<div
  class="toggle is-small is-inverse"
>
  <input
    checked=""
    id="Radio1"
    name="sample"
    type="radio"
    value="Radio1"
  />
  <label
    for="Radio1"
  >
    Radio 1
  </label>
  <input
    id="Radio2"
    name="sample"
    type="radio"
    value="Radio2"
  />
  <label
    for="Radio2"
  >
    Radio 2
  </label>
  <input
    id="Radio3"
    name="sample"
    type="radio"
    value="Radio3"
  />
  <label
    for="Radio3"
  >
    Radio 3
  </label>
</div>
`;

exports[`DnToggle Tests should render with custom props 1`] = `
<div
  class="toggle is-small is-inverse additional--class"
>
  <ul>
    <li
      class="is-active"
    >
      Toggle1
    </li>
    <li
      class="is-disabled"
    >
      Toggle2
    </li>
    <li
      class=""
    >
      Toggle3
    </li>
  </ul>
</div>
`;

exports[`DnToggle Tests should render with default props 1`] = `
<div
  class="toggle"
>
  <ul>
    <li
      class="is-active"
    >
      Toggle1
    </li>
    <li
      class=""
    >
      Toggle2
    </li>
  </ul>
</div>
`;

exports[`DnToggle Tests should render with no class names 1`] = `
<div
  class="toggle"
>
  <ul>
    <li
      class=""
    >
      Toggle1
    </li>
    <li
      class=""
    >
      Toggle2
    </li>
  </ul>
</div>
`;
